metricbeat/module/jolokia/jmx/config.go

```diff
@@ -37,8 +38,22 @@ type RequestBlock struct {
 	Attribute []string `json:"attribute"`
 }
 
-func buildRequestBodyAndMapping(mappings []JMXMapping) ([]byte, map[string]string, error) {
-	responseMapping := map[string]string{}
+type attributeMappingKey struct {
+	mbean, attr string
+}
+
+// AttributeMapping contains the mapping information between attributes in Jolokia
+// responses and fields in metricbeat events
+type AttributeMapping map[attributeMappingKey]Attribute
+
+// Get the mapping options for the attribute of an mbean
+func (m AttributeMapping) Get(mbean, attr string) (Attribute, bool) {
+	a, found := m[attributeMappingKey{mbean, attr}]
```

Comment:
Nit: Seems like you could return here directly.

Reply:
I think it doesn't work when the expression is a value obtained from a map.